src/swarm-lang/Swarm/Language/Parser/Value.hs

```diff
@@ -0,0 +1,148 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Parser for values of the Swarm language, indexed by type.
+--
+-- Note that this is completely separate from the main Swarm language
+-- parser in the other Swarm.Language.Parser.* modules.  Parsing
+-- *values* only ever happens at runtime (e.g. to implement the 'read'
+-- command), and does not need to deal with comments, source position
+-- tracking, etc.
```

Comment:
Why is a separate parser necessary? I do not object to having a simpler one as well, but it would be useful to have the technical problem with the full parser documented here. 🤔

Reply:
Hmmm.... good question, let me think about this for a bit.  The alternative, I suppose, would be to use the full parser but then have a separate step afterwards to take an arbitrary term and check whether it actually looks like a value of the proper type...

Reply:
Thanks for bringing this up, I reimplemented it using the full swarm-lang parser followed by checking the type and then ensuring the parsed term is a value.  This turns out to be a lot simpler and gives a much clearer pathway forward for eventually being able to handle things like delay, function, and command types.